### PR TITLE
(WSR 8) <Input/> - error and errorMessage

### PR DESCRIPTION
**Pull request by EgoziE**
- Props:
  - removed `units` use `<Input.Affix/>` component instead
  - removed `magnifyingGlass` use `<Search/>` component instead
  - removed `help` and `helpMessage` use a `<FormField/>` wrapper instead

- Testkit:
  - removed `clickUnit`
  - removed `getUnit`
  - removed `hasMagnifyingGlass`
  - removed `clickMagnifyingGlass`
  - removed `hasHelp`

### CODE REVIEW

File: src/MultiSelect/MultiSelect.driver.js [hunk 1]
```diff
@@ -23,9 +23,6 @@ const multiSelectDriverFactory = ({ element }) => {
     /** returns true if the input is focused */
     inputWrapperHasFocus: () => inputWrapper.classList.contains('hasFocus'),
 
-    /** returns true if the input has error */
-    inputWrapperHasError: () => inputWrapper.classList.contains('error'),
```
Comment (simast) [inline src/MultiSelect/MultiSelect.driver.js]: `MultiSelect.driver.d.ts`

File: src/MultiSelect/MultiSelect.js [hunk 2]
```diff
@@ -226,20 +219,8 @@ MultiSelect.propTypes = {
   /** passing `'select'`  will render a readOnly input with menuArrow suffix */
   mode: PropTypes.string,
 
-  /** Is input has errors
-   * @deprecated
-   * @see status
-   */
-  error: PropTypes.bool,
```
Comment (simast) [inline src/MultiSelect/MultiSelect.js]: `MultiSelect/index.d.ts`

File: src/MultiSelect/MultiSelect.uni.driver.js [hunk 3]
```diff
@@ -30,10 +30,6 @@ export const multiselectUniDriverFactory = (base, body) => {
     inputWrapperHasFocus: async () =>
       (await getInputWrapper()).classList.contains('hasFocus'),
 
-    /** returns true if the input has error */
-    inputWrapperHasError: async () =>
```
Comment (simast) [inline src/MultiSelect/MultiSelect.uni.driver.js]: `MultiSelect.uni.driver.d.ts`

File: src/Input/InputSuffix.js [hunk 4]
```diff
@@ -41,45 +33,23 @@ const InputSuffix = ({
...
   suffix,
-  focused,
   tooltipPlacement,
   onTooltipShow,
 }) => {
   const suffixes = [
     {
       component: () => (
-        <InputWarningSuffix
-          theme={theme}
```
[the remark this answers is not in the document]
Comment (EgoziE) [reply src/Input/InputSuffix.js]: The following PR is removing all theme related props/styles in input family. 👍

File: src/Input/Input.js [hunk 5]
```diff
@@ -226,15 +212,14 @@ class Input extends Component {
...
-              statusMessage={suffixStatusMessage}
+              status={hideStatusSuffix ? undefined : status}
+              statusMessage={statusMessage}
               theme={theme}
               disabled={disabled}
               onIconClicked={onIconClicked}
               isClearButtonVisible={isClearButtonVisible}
               onClear={this.handleSuffixOnClear}
               menuArrow={menuArrow}
-              focused={this.state.focus}
```
[the remark this answers is not in the document]
Comment (EgoziE) [reply src/Input/Input.js]: I couldn't figure out what is the meaning of giving focus to **any** InputSuffix, the only use-case is with clearButton that doesn't work anyway.
The only place clearButton works is `<Dropdown/>`

Anyway, if we ever want to fix this clearButton issue, we'll do it in a different PR unrelated to WSR8

File: src/Input/Input.js [hunk 6]
```diff
@@ -130,25 +128,13 @@ class Input extends Component {
...
-    if (error) {
-      suffixStatus = Input.StatusError;
-      suffixStatusMessage = errorMessage;
-    }
-
-    const hasErrors = suffixStatus === Input.StatusError;
-
     // this doesn't work for uncontrolled, "value" refers only to controlled input
     const isClearButtonVisible =
-      this._isClearFeatureEnabled && !!value && !hasErrors && !disabled;
```
[the remark this answers is not in the document]
Comment (EgoziE) [reply src/Input/Input.js]: I asked UX, this is OK for all status 🥇

File: src/MultiSelect/index.d.ts [hunk 7]
```diff
@@ -5,6 +5,8 @@ import InputWithOptions, {InputWithOptionsProps, OnSelectFnSignature} from '../I
 import {DropdownLayoutProps, DropdownLayoutValueOption} from '../DropdownLayout';
 import {TagProps} from '../Tag';
 
+export type StatusProps = 'error' | 'warning' | 'loading';
```
Comment (simast) [inline src/MultiSelect/index.d.ts]: `MultiSelectStatus` - note that this is reexported from main `src/index.d.ts` and is accessible to users

File: src/Input/Input.driver.d.ts [hunk 8]
```diff
@@ -43,13 +43,8 @@ export interface InputDriver extends BaseDriver {
   prefixComponentExists: (style: string) => boolean;
   suffixComponentExists: (style: string) => boolean;
   isMenuArrowLast: () => boolean;
-  hasExclamation: () => boolean;
   isNarrowError: () => boolean;
```
Comment (simast) [inline src/Input/Input.driver.d.ts]: This method seems to have been removed as well

File: src/Input/Input.uni.driver.d.ts [hunk 9]
```diff
@@ -43,10 +43,6 @@ export interface InputUniDriver extends BaseUniDriver {
   getReadOnly: () => Promise<any>;
   getDisabled: () => Promise<any>;
   getTextOverflow: () => Promise<string>;
-  hasExclamation: () => Promise<boolean>;
-  hasError: () => Promise<boolean>;
-  hasWarning: () => Promise<boolean>;
-  hasLoader: () => Promise<boolean>;
```
Comment (simast) [inline src/Input/Input.uni.driver.d.ts]: also let's remove `isNarrowError` form this file

**Comment by EgoziE** [comment]
I'll add them in another PR, I'm gonna rewrite all migration doc in a better way

**Pull request merged**